pyquil/paulis.py

```diff
@@ -97,6 +97,22 @@ def __len__(self):
         """
         return len(self._ops)
 
+    def copy(self):
+        """
+        Properly creates a new PauliTerm, with a completely new dictionary
+        of operators
+        """
+        new_term = PauliTerm("I", 0, 0)
+        # coefficient and _id are copied as new refs (numbers, strings immutable)
+        new_term.coefficient = self.coefficient
+        new_term._id = self._id
+        # _ops dict needs to use own copy method
+        new_term._ops = self._ops.copy()
```

Comment:
I'd prefer if we iterate over the `self.__dict__` attribute and copy its contents, while explicitly excluding attributes we don't want to incorporate. This will avoid having to remember to adjust this method if we add and delete items